### PR TITLE
New package: a4-0.2.1

### PR DESCRIPTION
**Pull request by littlewing7**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-musl)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl
  - armv7l
  - armv7l-musl
  - armv6l-musl

### CODE REVIEW

File: srcpkgs/a4/template [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Template file for 'a4'
+pkgname=a4
+version=0.2.1
+revision=1
+#archs="i686 x86_64"
```
Comment (abenson) [inline srcpkgs/a4/template:5]: Remove the commented lines.

Comment (littlewing7) [reply srcpkgs/a4/template:5]: I apologize, I'm new to this tasks! 

Commended lines removed!

Many thanks

Max

**Comment by littlewing7** [comment]
I apologize, for some reasons I don't remove all the commented lines. 

now should be ok! 

many thanks for you patience!!
Max

File: srcpkgs/a4/template (hunk 1 above)
Comment (littlewing7) [reply srcpkgs/a4/template]: I think to have cleaned correctly the template, can you please confirm_ 

many thanks 
Max

**Comment by littlewing7** [comment]
Hi Guys, 

this package doesn't break any other packages. 
There are any particular reason preventing this package to be merged?
Can I help ?

Many thanks for your kindly reply.

Max

File: srcpkgs/a4/template [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Template file for 'a4'
+pkgname=a4
+version=0.2.1
+revision=1
+build_style=gnu-makefile
+hostmakedepends="libtool pkg-config"
+makedepends="libvterm-devel libtickit-devel"
+depends="libvterm>=0.3.1 libtickit>=0.4.3"
```
Comment (Duncaen) [inline srcpkgs/a4/template:8]: Is it really required to manually define minimum versions here?

File: srcpkgs/a4/template [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+homepage="https://a4term.com/"
+distfiles="https://github.com/rpmohn/a4/archive/v${version}.tar.gz"
+checksum=52e16e58ac52b9157933a3fd1063bc79e8dfa2f72ea2b4f03e77327bbec2c59e
+
+post_extract() {
+	if [ "$CROSS_BUILD" ]; then
+		cp /usr/bin/libtool .
+		sed -e "s,CCLD=.*,CCLD=$CC,g;s,CC=.*,CC=$CC,g" -i libtool
+		sed -e "s,libtool,./libtool,g" -i Makefile
+	fi
```
Comment (Duncaen) [inline srcpkgs/a4/template:21]: This looks kinda wrong, why is this required?

File: srcpkgs/a4/template (hunk 2 above)
Comment (littlewing7) [reply srcpkgs/a4/template:8]: Hi Duncan, 

many thanks for your reply, regarding this i'm not sure if this is necessary inside the template, but it is a requirements of the original software: https://github.com/rpmohn/a4 (inside the readme I found  this requirements). 

before submitting this new package I already update the void templates for this required packages with this PRs: 
(https://github.com/void-linux/void-packages/pull/43532)
(https://github.com/void-linux/void-packages/pull/43658)

So i'm not sure if the line you mention now in this template is needed or not, please help me to understand

Many thanks

Comment (Duncaen) [reply srcpkgs/a4/template:8]: Defining `depends` manually will overwrite the default minimum version based on the shared library version. 

```
void-packages% cat common/shlibs|grep libvterm
libvterm.so.0 libvterm-0.3_1
void-packages% cat common/shlibs|grep libtickit
libtickit.so.3 libtickit-0.4.1_1
```

Just for simplicity and for future updates I prefer not to manually overwrite this.

**Comment by littlewing7** [comment]
Hi Duncan, 

can help me how to correctly define the template to obtain the right compilation of this new package without breaking the common/shlibs, and contemporary satisfy the requirements of the original software: https://github.com/rpmohn/a4 (inside the readme I found this requirements).

before submitting this new package I already update the void templates for this required packages with this PRs:
(https://github.com/void-linux/void-packages/pull/43532)
(https://github.com/void-linux/void-packages/pull/43658)

Many thanks

Massimo

File: srcpkgs/a4/template [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'a4'
+pkgname=a4
+version=0.2.1
+revision=1
+build_style=gnu-makefile
+hostmakedepends="libtool pkg-config"
+makedepends="libvterm-devel libtickit-devel"
+depends="libvterm libtickit"
```
Comment (Duncaen) [inline srcpkgs/a4/template:8]: Those `depends` are automatically detected through the shared library check and don't need to be specified.

File: srcpkgs/a4/template [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'a4'
+pkgname=a4
+version=0.2.1
+revision=1
+build_style=gnu-makefile
+makedepends="libvterm-devel libtickit-devel"
+short_desc="A4 is a dynamic terminal window manager, partial rewrite of dvtm"
```
Comment (abenson) [inline srcpkgs/a4/template:7]:
```suggestion
short_desc="Dynamic terminal window manager, partial rewrite of dvtm"
```

File: srcpkgs/a4/template [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'a4'
+pkgname=a4
+version=0.2.1
+revision=1
+build_style=gnu-makefile
+makedepends="libvterm-devel libtickit-devel"
+short_desc="Dynamic terminal window manager, partial rewrite of dvtm"
```
Comment (Duncaen) [inline srcpkgs/a4/template:7]: I would match the upstream short description:

> Dynamic terminal manager and multiplexer

**Comment by littlewing7** [comment]
Done 

Max
